lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -148,12 +148,12 @@ export abstract class ClientApplication {
              * Cache the hash to be retrieved after the next redirect
              */
             const hashKey = this.browserStorage.generateCacheKey(TemporaryCacheKeys.URL_HASH);
-            this.browserStorage.setItem(hashKey, responseHash, CacheSchemaType.TEMPORARY);
+            this.browserStorage.setTemporaryCache(hashKey, responseHash);
             if (!loginRequestUrl || loginRequestUrl === "null") {
                 // Redirect to home page if login request url is null (real null or the string null)
                 const homepage = BrowserUtils.getHomepage();
                 // Cache the homepage under ORIGIN_URI to ensure cached hash is processed on homepage
-                this.browserStorage.setItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.ORIGIN_URI), homepage, CacheSchemaType.TEMPORARY);
+                this.browserStorage.setTemporaryCache(this.browserStorage.generateCacheKey(TemporaryCacheKeys.ORIGIN_URI), homepage);
```

Comment:
Can we make `this.browserStorage.generageCacheKey` a step inside `setTemporaryCache` and just pass the `TemporaryCacheKeys.ORIGIN_URI` here?

Reply:
Done!